zerver/views/messages.py

```diff
@@ -300,13 +300,21 @@ def by_id(self, query: Query, operand: str, maybe_negate: ConditionTransform) ->
         cond = self.msg_id_column == literal(operand)
         return query.where(maybe_negate(cond))
 
-    def by_pm_with(self, query: Query, operand: str, maybe_negate: ConditionTransform) -> Query:
+    def by_pm_with(self, query: Query, operand: Union[str, Iterable[int]],
+                   maybe_negate: ConditionTransform) -> Query:
         # This will strip our own email out of the operand
         # and do other munging.
-        emails = raw_pm_with_emails(
-            email_str=operand,
-            my_email=self.user_profile.email,
-        )
+        if isinstance(operand, str):
+            emails = raw_pm_with_emails(
+                email_str=operand,
+                my_email=self.user_profile.email,
+            )
+        else:
+            emails = raw_pm_with_emails_by_ids(
```

Comment:
Let's add a comment here:
"""
This is where we handle passing a list of user IDs for the narrow, which is the 
preferred/cleaner API.  Ideally, we'd refactor the below to natively use user IDs, 
rather than having mapping user IDs to emails here.
"""